src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -815,31 +814,27 @@ private static void DisplaySearchPaths(TextWriter writer, List<string> attempted
 
             private async Task<ScriptState<object>> ExecuteOnUIThread(Script<object> script, ScriptState<object> stateOpt)
             {
-                return await Task.Factory.StartNew(async () =>
-                {
-                    try
-                    {
-                        var task = (stateOpt == null) ?
-                            script.RunAsync(_globals, CancellationToken.None) :
-                            script.ContinueAsync(stateOpt, CancellationToken.None);
-
-                        return await task.ConfigureAwait(false);
-                    }
-                    catch (FileLoadException e) when (e.InnerException is InteractiveAssemblyLoaderException)
-                    {
-                        Console.Error.WriteLine(e.InnerException.Message);
-                        return null;
-                    }
-                    catch (Exception e)
+                return await ((Task<ScriptState<object>>)s_control.Invoke(
+                    (Func<Task<ScriptState<object>>>)(() =>
                     {
-                        // TODO (tomat): format exception
-                        Console.Error.WriteLine(e);
-                        return null;
-                    }
-                },
-                CancellationToken.None,
-                TaskCreationOptions.None,
-                s_UIThreadScheduler).Unwrap().ConfigureAwait(false);
+                        try
+                        {
+                            return (stateOpt == null) ?
+                                script.RunAsync(_globals, CancellationToken.None) :
+                                script.ContinueAsync(stateOpt, CancellationToken.None);
+                        }
+                        catch (FileLoadException e) when (e.InnerException is InteractiveAssemblyLoaderException)
+                        {
+                            Console.Error.WriteLine(e.InnerException.Message);
+                            return null;
```

Comment:
Right you are...it was *not* redundant.  I've changed it to be ```Task.FromResult<ScriptState<object>>(null)```.  I don't think there's a singleton, but if there is, we can use that.  I personally kind of prefer this rather than the extra ```async``` indirection, because it's one less place to get ```ConfigureAwait``` wrong.